users/ericgebhart/extensions/encoders.c

```diff
@@ -0,0 +1,83 @@
+/*
+  Copyright 2022 Eric Gebhart <e.a.gebhart@gmail.com>
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifdef ENCODER_ENABLE
```

Comment:
Would likely be simpler to just use encoder map.

Reply:
Ah, yea. the encoder stuff changed on me.  I´ll take a look.

Reply:
Encoder map support was only added recently (the last breaking changes cycle, 0.17), so it is still pretty new.  Docs for it: 
https://docs.qmk.fm/#/feature_encoders?id=encoder-map

But myself and a few others use it, and it works well. Even supports custom keycodes. ;)

Reply:
I saw the change, read the doc when you reminded me, but haven´t refactored to it.  I´ll see what it takes. I like to support the internals when they exist.
probably tomorrow, I´m on Paris time.

Reply:
Ah, but now I see that I have more functionality than the new encoder maps.  I can specify a mod and layer which then causes the correct encoder choice.  encoder maps are only for a layer.  I have multiple encoder actions defined per layer depending on the mod being applied.

Reply:
perhaps I should add that to core so I can adopt it.

Reply:
> perhaps I should add that to core so I can adopt it.

You could use `process_record_user`, actually, since everything goes through that.  And if needed, you can filter with `IS_ENCODEREVENT` and/or `IS_KEYEVENT`.

But yeah, it's a bit different from your implementation, and mostly wanted to make sure you were aware of it because there is definitely a good deal of overlap, too.